frontend/src/actions/user.ts

```diff
@@ -0,0 +1,38 @@
+"use server";
+
+import { auth, fetchToAdonis } from "@/lib/auth";
+import type { UserSettingsPayload } from "@/types";
+
+export const getCurrentUser = async () => {
+  const user = await auth();
+  if (user == null) {
+    throw new Error("Not logged in");
+  }
+  return user;
+};
+
+interface UpdateUserSettingsResponse {
+  message: string;
+  user: number;
+  success: boolean;
+}
+
+export const updateUser = async (payload: UserSettingsPayload) => {
+  const user = await auth();
+  if (user == null) {
+    throw new Error("Not logged in");
+  }
+
+  // Update user
+  const data = await fetchToAdonis<UpdateUserSettingsResponse>({
+    url: "/user",
+    method: "PATCH",
+    body: JSON.stringify(payload),
+  });
+
+  if (data === null) {
```

Comment:
Zastanawiam się, czy może nie lepiej zamiast sypać errorami zwracać coś typu {  success: false, message: "Failed to update user" } i później nie trzeba będzie try catchy robić tylko if (success !== true){return <h2>message</h2>}
Ale to tylko taki pomysł